### ISSUE TITLE
toolbox:versions -DapplyToPom=true defines a new property in BOM instead of updating parent project

### ISSUE THREAD

**Issue by jakub-bochenski** [maveniverse/toolbox#154]
See also https://github.com/mojohaus/versions/issues/1200

I tried using toolbox as an alternative.  
It's better in that it detects that the property should be upgraded.

But it defines a new property in child/BOM module instead of updating the property defined in parent module.

**Comment by cstamas** [comment]
Yup, `applyToPom` is still limited, especially when a whole reactor is to be observed... is still marked as "experimental".

It works for simple use cases like having a parent POM that already follows "best practices" for now...

**Comment by jakub-bochenski** [comment]
> It works for simple use cases like having a parent POM that already follows "best practices" for now...

Care to elaborate what those best practices are? I thought my module structure already does follow best practices

**Comment by cstamas** [comment]
@jakub-bochenski could you put your reproducer into a git repo? It would be much easier to show.... but on a related note:
https://github.com/apache/maven-site/pull/598#discussion_r1894013277

**Comment by jakub-bochenski** [comment]
interesting, thanks!
so the gist of what I'm doing wrong seems to be:

> In short: only external BOMs should be imported to reactor.
